packages/exhibitor/extra/dcos_zk_backup.py

```diff
@@ -0,0 +1,322 @@
+    # while running, no data is lost inplace.
+    # However, this has a downside that we may use a lot of disk space -
+    # we have seen ZooKeeper files up to 20 GB in size.
+    _copy_dir_and_preserve_ownership(
+        src=zookeeper_dir,
+        dst=tmp_zookeeper_dir,
+        verbose=verbose,
+    )
+    shutil.rmtree(path=str(zookeeper_dir), ignore_errors=True)
+
+    print('Restoring {zookeeper_dir} from backup {backup}'.format(
+        zookeeper_dir=zookeeper_dir,
+        backup=backup,
+    ))
+    with tarfile.open(name=backup, mode='r:gz') as tar:
+        tar.extractall(path=str(EXHIBITOR_DIR))
+        if verbose:
+            tar.list()
+
+    print('Deleting temporary files in {tmp_zookeeper_dir}'.format(
+        tmp_zookeeper_dir=tmp_zookeeper_dir,
+    ))
+    shutil.rmtree(path=str(tmp_zookeeper_dir), ignore_errors=True)
+
+    print('Local ZooKeeper instance restored successfully')
+
+
+def _non_existing_file_path_existing_parent_dir(value) -> Path:
+    """
+    Validate that the value is a path to a file which does not exist
+    but the parent directory tree exists.
+    """
+    path = Path(value)
+    if path.exists():
+        raise ArgumentTypeError('Error: {} already exists'.format(path))
+    if not Path(path.parent).exists():
+        raise ArgumentTypeError(
+            'Error: {} parent directory does not exist'.format(path),
+        )
+    return path.absolute()
+
+
+def _existing_file_path(value) -> Path:
+    """
+    Validate that the value is a file which does exist on the file system.
+    """
+    path = Path(value)
+    if not path.exists():
+        raise ArgumentTypeError('Error: {} does not exist'.format(path))
+    if not path.is_file():
+        raise ArgumentTypeError('Error: {} is not a file'.format(path))
+    return path.absolute()
+
+
+def _existing_dir_path(value) -> Path:
+    """
+    Validate that the value is a directory which does exist on the file system.
+    """
+    path = Path(value)
+    if not path.exists():
+        raise ArgumentTypeError('Error: {} does not exist'.format(path))
+    if not path.is_dir():
+        raise ArgumentTypeError('Error: {} is not a directory'.format(path))
+    return path.absolute()
+
+
+class DCOSZooKeeperCli:
+    """
+    Minimal CLI to backup/restore the local DC/OS ZooKeeper instance.
+    """
+
+    def __init__(self):
+        """
+        Present CLI command choices.
+        """
+        parser = ArgumentParser(
+            description=(
+                'Command line utility to more safely interact with the local '
+                'ZooKeeper instance on DC/OS master nodes'
+            )
+        )
+        parser.add_argument(
+            'command',
+            type=str,
+            choices=[
+                'backup',
+                'restore',
+            ],
+            help="CLI commands available",
+        )
+        args = parser.parse_args(sys.argv[1:2])
+        getattr(self, args.command)()
```

Comment:
Argument parser library supports subcommands - https://docs.python.org/3/library/argparse.html#sub-commands

Reply:
I've read up and tried `argparse` `subcommands` and I think the current way is a superior solution, especially if we ever want add more commands.
After manually digging into a transaction log I can totally see a `dcos-zk open logfile` command that we should implement.